lib/vaahextendflutter/base/base_controller.dart

```diff
@@ -42,6 +43,9 @@ class BaseController extends GetxController {
       // RootAssets
       Get.put(RootAssetsController());
 
+      // DataDog
+      await DataDogLoggingService.init();
```

Comment:
`we should not directly use DataDogLoggingService.`

Instead of calling classes directly from DataDogLoggingService, we should be achieving logging through abstraction and using the Log class we created (Log > Remote > Datadog | Sentry), calling logs from this main class makes sure that the developer doesn't have to change all the instances of calls we make direct using DataDogLoggingService - when project decides to switch the service, and calling one function from Log (main) class will ensure that it just doesn't log the logs on Datadog but on other services we integrated (i.e. Sentry & Local Logging).

Reply:
As singleton class is used all over the services, each services are having different different params and properties based on this if we want to go for approach of abstraction then we need to rewrite entire code for Sentry and Datadog services. Instead of this we can go for SingleResponsibility Principle so everything will be independent from each other. This may help us in future.

Reply:
I still think we need to use abstraction & inheritance. We can connect & you can explain over there?